### PR TITLE
bpo-18299: Improving eintrdata's test suites

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue18299

### CODE REVIEW

File: Lib/test/support/script_helper.py [hunk 1]
```diff
@@ -122,11 +122,14 @@ def run_python_until_end(*args, **env_vars):
...
         env['TERM'] = ''
 
+    universal_newlines = env_vars.pop('__universal_newlines', False)
+
     env.update(env_vars)
     cmd_line.extend(args)
     proc = subprocess.Popen(cmd_line, stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                         env=env, cwd=cwd)
+                         env=env, cwd=cwd,
```
Comment (mangrisano) [inline Lib/test/support/script_helper.py]: I think this line and the over one should be indented under `cmd_line`.

File: Lib/test/support/script_helper.py [hunk 2]
```diff
@@ -122,11 +122,14 @@ def run_python_until_end(*args, **env_vars):
...
+                         universal_newlines=universal_newlines)
```
Comment (mangrisano) [inline Lib/test/support/script_helper.py]: Same thing as over.

File: Lib/test/support/script_helper.py (hunk 1 above)
Comment (shihai1991) [reply Lib/test/support/script_helper.py]: done:)

File: Lib/test/support/script_helper.py (hunk 2 above)
Comment (shihai1991) [reply Lib/test/support/script_helper.py]: done:)

**Comment by mangrisano** [comment]
Did you test this change? If not, please provide it.

**Comment by mangrisano** [comment]
Furthermore, you also need to fix the doc adding the parameter in the signature.

**Comment by shihai1991** [comment]
> Did you test this change? If not, please provide it.
Let us waiting other developer's opinion, if they think this patch is fine, i will replenish it;)

**Comment by shihai1991** [comment]
@vstinner Hi, Victor. Looks i choose the complicated way of script_helper. Maybe i could find the way which you said to supply popen more easily.

**Comment by gpshead** [comment]
For a new feature in `test.support.script_helper` it would help to have some motivating examples refactoring some existing test suite to cleaner due to the new API.

**Comment by shihai1991** [comment]
@gpshead Thanks for review. According your and victor's opinion, I refactor the eintr_tester's test suites.
And pls forget my previous patch;)

File: Lib/test/eintrdata/eintr_tester.py [hunk 3]
```diff
@@ -71,18 +71,14 @@ def tearDown(self):
...
-
 
 @unittest.skipUnless(hasattr(signal, "setitimer"), "requires setitimer()")
 class OSEINTRTest(EINTRBaseTest):
     """ EINTR tests for the os module. """
 
     def new_sleep_process(self):
         code = 'import time; time.sleep(%r)' % self.sleep_time
-        return self.subprocess(code)
+        return spawn_python('-c', code)
```
Comment (gpshead) [inline Lib/test/eintrdata/eintr_tester.py:81]: I don't see how this is beneficial.  The point of that method is to avoid having to type '-c' everywhere.

**Comment by gpshead** [comment]
This doesn't materially improve the eintrdata test suite and the issue it links to is about script_helper, _assert_python, and text mode to the child.  which don't seem to be a problem in eintrdata_tester.py

File: Lib/test/eintrdata/eintr_tester.py (hunk 3 above)
Comment (shihai1991) [reply Lib/test/eintrdata/eintr_tester.py:81]: > This doesn't materially improve the eintrdata test suite and the issue it links to is about script_helper, _assert_python, and text mode to the child. which don't seem to be a problem in eintrdata_tester.py

ok, the spawn_python in script_helper is good enough. Forget this.

**Pull request closed without merging**